pkg/policy/repository.go

```diff
@@ -391,6 +392,18 @@ func (p *Repository) computePolicyEnforcementAndRules(securityIdentity *identity
 		}
 	}
 
+	// Always sort matched rules to get a stable policy order.
+	// It's not the order per se that is important, just that it's always in the same order when the
+	// elements are the same. In most cases this should be a small list so the overhead should be pretty minimal.
+	// This is very useful for subsystems like the dnsproxy that can reuse the same regex during recompilation
+	// if the list of FQDNs is the same.
+	slices.SortFunc(matchingRules, func(a, b *rule) int {
+		if sign := cmp.Compare(a.key.resource, b.key.resource); sign != 0 {
```

Comment:
Since we only need the ordering to be stable, should we compare `idx` first instead of `resource`?
`idx` is an `uint`, so its comparison is much cheaper than comparing the resource `string`.